tools/KoreBuild.Console/Commands/MSBuildCommand.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.CommandLineUtils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace KoreBuild.Console.Commands
+{
+    internal class MSBuildCommand : SubCommandBase
+    {
+        private bool EnableBinaryLog => Environment.GetEnvironmentVariable("KOREBUILD_ENABLE_BINARY_LOG") == "1";
+
+        private List<string> Arguments { get; set; }
+
+        public override void Configure(CommandLineApplication application)
+        {
+            base.Configure(application);
+            Arguments = application.RemainingArguments;
+        }
+
+        protected override int Execute()
+        {
+            Log($"Building {RepoPath}.");
+            Log($"dotnet = {RepoPath}");
+
+            if(SDKVersion != "latest")
+            {
+                var globalFile = Path.Combine(RepoPath, "global.json");
+                File.WriteAllText(globalFile, $"{{ \"sdk\": {{ \"version\": \"{SDKVersion}\" }} }}", System.Text.Encoding.ASCII);
+            }
+            else
+            {
+                Log($"Skipping global.json generation because the SDKVersion = {SDKVersion}");
+            }
+
+            var makeFileProj = Path.Combine(KoreBuildDir, "KoreBuild.proj");
+            var msBuildArtifactsDir = Path.Combine(RepoPath, "artifacts", "msbuild");
+            var msBuildResponseFile = Path.Combine(msBuildArtifactsDir, "msbuild.rsp");
+
+            var msBuildLogArgument = string.Empty;
+
+
+            if(EnableBinaryLog)
+            {
+                Log("Enabling binary logging");
+                var msBuildLogFilePath = Path.Combine(msBuildArtifactsDir, "msbuild.binlog");
+                msBuildLogArgument = $"/bl:{msBuildLogFilePath}";
+            }
+
+            var msBuildArguments = string.Empty;
+
+            foreach (var arg in Arguments)
+            {
+                msBuildArguments += Environment.NewLine + arg;
+            }
+
+            msBuildArguments += $@"
```

Comment:
This is may be problematic for Linux...we should test. Literal multiline strings in C# will capture `\r\n` in its value. `dotnet msbuild @file.rsp` may be expecting only `\n` on Linux.

To test this, build KoreBuild on Windows and run the artifact on Linux.